types/index.d.ts

```diff
@@ -1981,6 +1979,8 @@ export declare namespace Knex {
 
   interface ReferencingColumnBuilder extends ColumnBuilder {
     inTable(tableName: string): ColumnBuilder;
+    deferrable(type: deferrableType): ColumnBuilder;
+    withKeyName(keyName: string): ColumnBuilder;
```

Comment:
Put This two in ```ReferencingColumnBuilder``` instead of just ```ForeignConstraintBuilder & MultikeyForeignConstraintBuilder``` since they need to be both accessible through ```references``` and ```foreign``` method